app/models/miq_expression.rb

```diff
@@ -1557,6 +1557,30 @@ def self._quick_search?(e)
     end
   end
 
+  def custom_attribute_columns
+    exp ? self.class._custom_attribute_columns(exp) : []
+  end
+
+  def self._custom_attribute_columns(expression)
+    result = []
+    case expression
+    when Array
+      expression.each { |x| result.concat(_custom_attribute_columns(x)) }
```

Comment:
@kbrock that's nice, I just used `flat_map` instead of `collect`, because there were array with nested arrays (like `[[[]], [],.. ]`) but thanks!